fps/src/main/scala/chapter03/exercise3_3.scala

```diff
@@ -0,0 +1,38 @@
+package chapter03
+
+import chapter03.exercise3_1.List.sum
+import chapter03.exercise3_2.List
+
+object exercise3_3 extends App {
+  sealed trait List[+A]
+  case object Nil extends List[Nothing]
+  case class Cons[+A](head: A, tail: List[A]) extends List[A]
+
+  object List {
+    def sum(ints: List[Int]): Int = ints match {
+      case Nil => 0
+      case Cons(x, xs) => x + sum(xs)
+    }
+
+    def product(ds: List[Double]): Double = ds match {
+      case Nil => 1.0
+      case Cons(0.0, _) => 0.0
+      case Cons(x, xs) => x * product(xs)
+    }
+    def setHead[A](ds: List[A], newVal: A): List[A] = {
```

Comment:
fyi - doesn't really "sets" the head, but returns a _new_ list with the head concatenated which is why this function normally is named `concat`